File: src/main/java/com/optica/controllers/ApplyController.java
package com.optica.controllers;

import java.time.LocalDate;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.ResponseBody;

import com.optica.domain.Apply;
import com.optica.domain.Curriculum;
import com.optica.entity.User;
import com.optica.services.ApplyService;
import com.optica.services.login.SecurityService;

@Controller
public class ApplyController {

	@Autowired
	private ApplyService applyService;

	@Autowired
	private SecurityService securityService;

	@PostMapping("/apply/{jobOfferId}")
	@ResponseBody
	public String applyForJobOffer(Model model, Long jobOfferId) {

		Apply apply = new Apply();
		User user = securityService.getCurrentUser();

		if (user.getCurriculum() != null) {
			apply.setActive(true);
			apply.setApplyDate(LocalDate.now());
			apply.setCurriculumId(user.getCurriculum().getId());
			apply.setJobOfferId(jobOfferId);
			applyService.applyForJobOffer(apply);
		}

		return "success";
	}

}

File: src/main/kotlin/com/optica/controllers/ApplyController.kt
package com.optica.controllers

import java.time.LocalDate
import java.util.Optional

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.ResponseBody

import com.optica.domain.Apply
import com.optica.domain.Curriculum
import com.optica.entity.User
import com.optica.services.ApplyService
import com.optica.services.login.SecurityService

@Controller
class ApplyController {

    @Autowired
    lateinit  var applyService: ApplyService

    @Autowired
    lateinit  var securityService: SecurityService

    @PostMapping("/apply/{jobOfferId}")
    @ResponseBody
    fun applyForJobOffer(model: Model, jobOfferId: Long?): String {

        val apply = Apply()
        val user = securityService!!.currentUser

        if (user.curriculum != null) {
            apply.active = true
            apply.applyDate = LocalDate.now()
            apply.curriculumId = user.curriculum!!.id
            apply.jobOfferId = jobOfferId
            applyService!!.applyForJobOffer(apply)
        }

        return "success"
    }

}
